File: BackEnd/src/Shared/Ashraak.SharedKernel.Contracts/CctvCrm/CctvFeatureFlags.cs
namespace Ashraak.SharedKernel.Contracts.CctvCrm;

/// <summary>
/// Feature flag keys for Aarvii CCTV business modules.
/// Evaluated via platform <see cref="FeatureFlags.Interfaces.IFeatureFlagService"/> (config section <c>Features:Flags</c>).
/// </summary>
public static class CctvFeatureFlags
{
    /// <summary>Master gate for all CCTV APIs and UI (health always available when registered).</summary>
    public const string Enabled = "cctv.enabled";

    public const string LeadsEnabled = "cctv.leads.enabled";
    public const string CustomersEnabled = "cctv.customers.enabled";
    public const string AmcEnabled = "cctv.amc.enabled";
    public const string ServiceEnabled = "cctv.service.enabled";
    public const string TicketsEnabled = "cctv.tickets.enabled";
    public const string EngineersEnabled = "cctv.engineers.enabled";
    public const string InvoicesEnabled = "cctv.invoices.enabled";
    public const string ReportingEnabled = "cctv.reporting.enabled";
    public const string CustomerPortalEnabled = "cctv.portal.customer.enabled";
    public const string EngineerPortalEnabled = "cctv.portal.engineer.enabled";
    public const string CustomerMobileEnabled = "cctv.mobile.customer.enabled";
    public const string EngineerMobileEnabled = "cctv.mobile.engineer.enabled";
    public const string SmsEnabled = "cctv.integrations.sms.enabled";
    public const string PdfEnabled = "cctv.integrations.pdf.enabled";

    /// <summary>All module flag keys for seeding config and documentation.</summary>
    public static readonly IReadOnlyList<string> All =
    [
        Enabled,
        LeadsEnabled,
        CustomersEnabled,
        AmcEnabled,
        ServiceEnabled,
        TicketsEnabled,
        EngineersEnabled,
        InvoicesEnabled,
        ReportingEnabled,
        CustomerPortalEnabled,
        EngineerPortalEnabled,
        CustomerMobileEnabled,
        EngineerMobileEnabled,
        SmsEnabled,
        PdfEnabled
    ];
}

File: FrontEnd/apps/web/src/modules/cctv/featureFlags/cctvFeatureFlags.ts
/**
 * CCTV feature flag keys — must stay in sync with
 * `Ashraak.SharedKernel.Contracts.CctvCrm.CctvFeatureFlags` (backend).
 *
 * Defaults are placeholders until modules ship (B1–B7). Override via
 * `VITE_CCTV_FEATURE_*` env vars or future tenant flag API.
 */
export const CCTV_FEATURE_FLAGS = {
  enabled: 'cctv.enabled',
  leads: 'cctv.leads.enabled',
  customers: 'cctv.customers.enabled',
  amc: 'cctv.amc.enabled',
  service: 'cctv.service.enabled',
  tickets: 'cctv.tickets.enabled',
  engineers: 'cctv.engineers.enabled',
  invoices: 'cctv.invoices.enabled',
  reporting: 'cctv.reporting.enabled',
  customerPortal: 'cctv.portal.customer.enabled',
  engineerPortal: 'cctv.portal.engineer.enabled',
  customerMobile: 'cctv.mobile.customer.enabled',
  engineerMobile: 'cctv.mobile.engineer.enabled',
  sms: 'cctv.integrations.sms.enabled',
  pdf: 'cctv.integrations.pdf.enabled',
} as const;

export type CctvFeatureFlagKey = (typeof CCTV_FEATURE_FLAGS)[keyof typeof CCTV_FEATURE_FLAGS];

/** Sprint 0 placeholder defaults — master on; module flags off until phase ships. */
export const CCTV_FEATURE_FLAG_DEFAULTS: Readonly<Record<CctvFeatureFlagKey, boolean>> = {
  [CCTV_FEATURE_FLAGS.enabled]: true,
  [CCTV_FEATURE_FLAGS.leads]: true,
  [CCTV_FEATURE_FLAGS.customers]: true,
  [CCTV_FEATURE_FLAGS.amc]: true,
  [CCTV_FEATURE_FLAGS.service]: true,
  [CCTV_FEATURE_FLAGS.tickets]: true,
  [CCTV_FEATURE_FLAGS.engineers]: true,
  [CCTV_FEATURE_FLAGS.invoices]: true,
  [CCTV_FEATURE_FLAGS.reporting]: true,
  [CCTV_FEATURE_FLAGS.customerPortal]: true,
  [CCTV_FEATURE_FLAGS.engineerPortal]: true,
  [CCTV_FEATURE_FLAGS.customerMobile]: true,
  [CCTV_FEATURE_FLAGS.engineerMobile]: true,
  [CCTV_FEATURE_FLAGS.sms]: false,
  [CCTV_FEATURE_FLAGS.pdf]: false,
};
